spyder/plugins/layout/plugin.py

```diff
@@ -146,6 +148,43 @@ def on_toolbar_available(self):
             before=PreferencesActions.Show
         )
 
+    @on_plugin_teardown(plugin=Plugins.MainMenu)
+    def on_main_menu_teardown(self):
+        mainmenu = self.get_plugin(Plugins.MainMenu)
+        # Remove Panes related actions from the View application menu
+        panes_items = [
+            "plugins_menu",

```

Comment:
Please add an enum to identify this menu so we don't have to reference it directly by its id here.